Week-4/Class.js

```diff
@@ -0,0 +1,48 @@
+//! ES6 Based Inheritance using Class
+
+class person {
+    constructor(name) {
+        this.name = name;
+    }
+    printName(){
+        console.log(this.name)
+    }
+}
+class Teacher extends person{
+    constructor(name,subject){
+        super(name)
+        this.subject = subject
+    }
+    teach(){
+        return console.log(this.name + ' is now teaching ' +this.subject)
+    }
+}
+const teacher = new Teacher('Dharmik','Maths');
+teacher.teach();
+teacher.printName();
+
+
+
+
+//! ES5 Based Inheritance using function
+
+function PERSON(NAME){
+    this.NAME = NAME;
+}
+PERSON.prototype.printname = function(){
+    console.log(this.NAME)
+}
+function TEACHER(NAME,SUBJECT){
+            PERSON.call(this, NAME);
+            this.SUBECT = SUBJECT;

```

Comment:
Parameters passed to a function should always be in camelcase like firstName not FirstName or even FIRST_NAME